orchestrator/workflows/steps.py

```diff
@@ -10,30 +10,90 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from copy import deepcopy
+from typing import Optional
+
+import structlog
 
 from orchestrator.db import db
 from orchestrator.db.models import ProcessSubscriptionTable
 from orchestrator.domain.base import SubscriptionModel
+from orchestrator.services.subscriptions import get_subscription
 from orchestrator.targets import Target
 from orchestrator.types import State, SubscriptionLifecycle, UUIDstr
+from orchestrator.utils.json import to_serializable
 from orchestrator.workflow import Step, step
 
+logger = structlog.get_logger(__name__)
+
 
 @step("Unlock subscription")
 def resync(subscription: SubscriptionModel) -> State:
+    """Transition a subscription to in sync."""
     subscription.insync = True
     return {"subscription": subscription}
 
 
 @step("Lock subscription")
-def unsync(subscription_id: UUIDstr) -> State:
-    subscription = SubscriptionModel.from_subscription(subscription_id)
-
+def unsync(subscription_id: UUIDstr, __old_subscriptions__: Optional[dict] = None) -> State:
+    """
+    Transition a subscription to out of sync.
+
+    This step will also create a backup of the current subscription details in the state with the key
+    `__old_subscriptions__`
+
+    Note:  This step will NOT overwrite any existing data in the state for the `__old_subscriptions__` key. Some
+    workflows already change subscription details in the initial form step. This is not a best practice but
+    it can be handy/needed for some scenarios. To ensure that you get a correct backup these forms need to create
+    the backup in the initial form step and save it to `__old_subscriptions__`.
+
+    An example, showing the creation of a backup for 2 subscriptions during the initial form step :
+
+    ```
+    subscription = YOUR_DOMAIN_MODEL.from_subscription(subscription_id)
+    subscription_backup = {}
+    subscription_backup[subscription.subscription_id] = deepcopy(subscription.dict())
+    subscription_backup[second_subscription.subscription_id] = deepcopy(second_subscription.dict())
+    return {..., "__old_subscriptions__": subscription_backup }
+    ```
+
+    It's also possible to make the backup for a subscription that doesn't have a domain model:
+
+    ```
+    from orchestrator.utils.json import to_serializable
+    subscription = subscriptions.get_subscription(subscription_id)
+    subscription_backup = {subscription_id: to_serializable(subscription)}
+    Do your changes here ...
+    return {..., "subscription": to_serializable(subscription), "__old_subscriptions__": subscription_backup }
+    ```
+
+    """
+    try:
+        subscription = SubscriptionModel.from_subscription(subscription_id)
+    except:  # noqa
```

Comment:
This can at least be `except Exception:`, right?

Reply:
As this is is at the core a Pydantic model we should get at least an instance of the Basic Pydantic exception? all other exceptions may fail, hard. IT should catch the ` PydanticTypeError` or the ` PydanticValueError`

Reply:
but probably a ValidationError wil do.